### PR TITLE
Add Gradle add/delete dependency util method

### PR DESCRIPTION
**Pull request by lucasmippon**
Relate https://github.com/jhipster/jhipster-lite/issues/1301

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/generator/buildtool/gradle/domain/GradleService.java [hunk 1]
```diff
@@ -1,11 +1,15 @@
 package tech.jhipster.lite.generator.buildtool.gradle.domain;
 
+import tech.jhipster.lite.generator.buildtool.generic.domain.Dependency;
 import tech.jhipster.lite.generator.project.domain.Project;
 
 public interface GradleService {
 
   void initJava(Project project);
 
+
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/buildtool/gradle/domain/GradleService.java:10]: strange, prettier should detect this

File: src/test/java/tech/jhipster/lite/generator/buildtool/gradle/domain/gradle/GradleDomainServiceTest.java [hunk 2]
```diff
@@ -55,4 +58,15 @@ void shouldAddGradleWrapper() {
...
+
+  @Test
+  void shouldAddDependency() {
+    Project project = tmpProject();
+    Dependency dependency = Dependency.builder().groupId("org.springframework.boot").artifactId("spring-boot-starter").build();
+
+    gradleDomainService.addDependency(project, dependency);
+
+    String newTextExpected = ("implementation(\"org.springframework.boot:spring-boot-starter\")\n" + GRADLE_NEEDLE_DEPENDENCY).indent(2);
+    verify(projectRepository).replaceText(project, "", "build.gradle.kts",REGEXP_SPACE_STAR + GRADLE_NEEDLE_DEPENDENCY, newTextExpected);
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/generator/buildtool/gradle/domain/gradle/GradleDomainServiceTest.java:70]: same, prettier should detect this format:
```suggestion
    verify(projectRepository).replaceText(project, "", "build.gradle.kts", REGEXP_SPACE_STAR + GRADLE_NEEDLE_DEPENDENCY, newTextExpected);
```

File: src/main/java/tech/jhipster/lite/generator/buildtool/gradle/domain/GradleService.java (hunk 1 above)
Comment (lucasmippon) [reply src/main/java/tech/jhipster/lite/generator/buildtool/gradle/domain/GradleService.java]: Look like prettier didn't detect it why I ran `npm run prettier:format`

File: src/test/java/tech/jhipster/lite/generator/buildtool/gradle/domain/gradle/GradleDomainServiceTest.java (hunk 2 above)
Comment (lucasmippon) [reply src/test/java/tech/jhipster/lite/generator/buildtool/gradle/domain/gradle/GradleDomainServiceTest.java]: Same thing, prettier didn't corrected it when I ran `npm run prettier:format`

Comment (pascalgrimaud) [reply src/test/java/tech/jhipster/lite/generator/buildtool/gradle/domain/gradle/GradleDomainServiceTest.java]: Yes there is a bug with prettier, I need to open a ticket to track it

**Comment by lucasmippon** [comment]
@pascalgrimaud do you know how I can have more information on Sonar analysis ? In CI it says there are two code smells.

**Comment by pascalgrimaud** [comment]
Yes @lucasmippon : the instructions to launch sonar is in the README

**Comment by lucasmippon** [comment]
@pascalgrimaud It look like there is an error when loading the uploader for codecov in CI

**Comment by pascalgrimaud** [comment]
@lucasmippon : yes it can happen, I'll restart the CI

**Comment by pascalgrimaud** [comment]
thanks a lot for your work @lucasmippon, I'll do my best to review it soon

**Pull request merged**